server/src/main/java/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -1186,6 +1227,431 @@ protected void pushToUpdateQueue(final List<NetworkVO> networks) throws Interrup
+                }
+            } catch (final Exception ex) {
+                s_logger.error("Fail to complete the FetchRouterHealthChecksResultTask! ", ex);
+                ex.printStackTrace();
+            }
+        }
+    }
+
+    private Map<String, Map<String, RouterHealthCheckResultVO>> getHealthChecksFromDb(long routerId) {
+        List<RouterHealthCheckResultVO> healthChecksList = routerHealthCheckResultDao.getHealthCheckResults(routerId);
+        Map<String, Map<String, RouterHealthCheckResultVO>> healthCheckResults = new HashMap<>();
+        if (healthChecksList.isEmpty()) {
+            return healthCheckResults;
+        }
+
+        for (RouterHealthCheckResultVO healthCheck : healthChecksList) {
+            if (!healthCheckResults.containsKey(healthCheck.getCheckType())) {
+                healthCheckResults.put(healthCheck.getCheckType(), new HashMap<>());
+            }
+            healthCheckResults.get(healthCheck.getCheckType()).put(healthCheck.getCheckName(), healthCheck);
+        }
+
+        return healthCheckResults;
+    }
+
+    private RouterHealthCheckResultVO updateRouterConnectivityHealthCheck(final long routerId, boolean connected, String message) {
+        boolean newEntry = false;
+        RouterHealthCheckResultVO connectivityVO = routerHealthCheckResultDao.getRouterHealthCheckResult(routerId, "connectivity", "basic");
+        if (connectivityVO == null) {
+            connectivityVO = new RouterHealthCheckResultVO(routerId, "connectivity", "basic");
+            newEntry = true;
+        }
+
+        connectivityVO.setCheckResult(connected);
+        connectivityVO.setLastUpdateTime(new Date());
+        connectivityVO.setCheckDetails(StringUtils.isNotEmpty(message) ? message.getBytes(Charset.forName("US-ASCII")) : null);
+
+        if (newEntry) {
+            routerHealthCheckResultDao.persist(connectivityVO);
+        } else {
+            routerHealthCheckResultDao.update(connectivityVO.getId(), connectivityVO);
+        }
+
+        return routerHealthCheckResultDao.getRouterHealthCheckResult(routerId, "connectivity", "basic");
+    }
+
+    private List<RouterHealthCheckResult> parseAndMergeDbHealthChecks(final long routerId, final String monitoringResult) {
+        if (StringUtils.isBlank(monitoringResult)) {
+            s_logger.warn("Attempted parsing empty monitoring results string for router " + routerId);
+            return Collections.emptyList();
+        }
+
+        try {
+            s_logger.info("Parsing and updating DB health check data for router: " + routerId);
+            s_logger.info("Retrieved data" + monitoringResult);
+            final Map<String, Map<String, RouterHealthCheckResultVO>> checksInDb = getHealthChecksFromDb(routerId);
+            final Type t = new TypeToken<Map<String, Map<String, Map<String, String>>>>() {}.getType();
+            final Map<String, Map<String, Map<String, String>>> checks = GsonHelper.getGson().fromJson(monitoringResult, t);
```

Comment:
Given the antiquated version is heavily used in code, I would prefer to keep the updating of Gson as a separate task since this feature is large enough with several minor fixes on Router monitoring. Thoughts?

Reply:
I chekced the current version and this call/construct is still supported so 👍

Reply:
and yes, the upgrade is definitely a different issue